dotCMS/src/main/java/com/dotmarketing/beans/MultiTree.java

```diff
@@ -30,97 +32,166 @@
     private int treeOrder;
 
     /** full constructor */
-
-    public MultiTree(String parent1, String parent2, String child, java.lang.String relationType, int treeOrder) {
-
-        this.parent1 = parent1;
-        this.parent2 = parent2;
+    public MultiTree(String htmlPage, String container, String child, String relationType, int treeOrder) {
+        this.parent1 = htmlPage;
+        this.parent2 = container;
         this.child = child;
-        this.relationType = relationType;
-        this.treeOrder = treeOrder;
+        this.relationType = (relationType == null) ? LEGACY_RELATION_TYPE : relationType;
+        this.treeOrder = (treeOrder < 0) ? 0 : treeOrder;
     }
 
     /** default constructor */
     public MultiTree() {
-
-        this.parent1 = "";
-        this.parent2 = "";
-        this.child = "";
+        this(null, null, null, LEGACY_RELATION_TYPE, 0);
     }
 
     /** minimal constructor */
 
-    public MultiTree(String parent1, String parent2,  String child) {
-        this.parent1 = parent1;
-        this.parent2 = parent2;
-        this.child = child;
+    public MultiTree(String htmlPage, String container, String child) {
+        this(htmlPage, container, child, LEGACY_RELATION_TYPE, 0);
     }
 
+    /**
+     * 
+     * {@link #getContentlet()}
+     */
+    @Deprecated
     public String getChild() {
+        return getContentlet();
+    }
+
+    public String getContentlet() {
         return this.child;
     }
 
-    public void setChild(String child) {
-        this.child = child;
+
+    @Deprecated
+    public MultiTree setChild(String child) {
+        return setContentlet(child);
     }
+
     public java.lang.String getRelationType() {
         return this.relationType;
     }
 
-    public void setRelationType(java.lang.String relationType) {
-        this.relationType = relationType;
+    public MultiTree setContentlet(String contentlet) {
+        this.child = contentlet;
+        return new MultiTree(this.parent1, this.parent2, contentlet, this.relationType, this.treeOrder);
     }
+
+    public MultiTree setRelationType(java.lang.String relationType) {
+        this.relationType = (relationType == null) ? LEGACY_RELATION_TYPE : relationType;
+        return new MultiTree(this.parent1, this.parent2, this.child, relationType, this.treeOrder);
+    }
+
     public int getTreeOrder() {
         return this.treeOrder;
     }
 
-    public void setTreeOrder(int treeOrder) {
+    public MultiTree setTreeOrder(int treeOrder) {
+        treeOrder = (treeOrder < 0) ? 0 : treeOrder;
         this.treeOrder = treeOrder;
+        return new MultiTree(this.parent1, this.parent2, this.child, this.relationType, treeOrder);
     }
 
     public String toString() {
         return ToStringBuilder.reflectionToString(this);
     }
 
     public boolean equals(Object other) {
-        if ( !(other instanceof MultiTree) ) return false;
+        if (!(other instanceof MultiTree))
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Compare to "this.getClass()" instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2162)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Missing curly brace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00121)